MyExample/WaitForm.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace MyExample
+{
+    public partial class WaitForm : Form
+    {
+        
+        private string _connection;
+        public WaitForm(string connection)
+        {
+            InitializeComponent();
+
+            this._connection = connection;
+            
+        }
+
+        private void timer1_Tick(object sender, EventArgs e)
+        {
+            progressBar1.Maximum = 20;
+            progressBar1.PerformStep();
+            if (progressBar1.Value == 20)
+            {
+                timer1.Enabled = false;
+                var fr = new FormRank(_connection);
```

Comment:
WaitForm should be a standalone component and should not initialize other components.
Try to use like:
Make an instance of WaitForm and when it finishes, initialize the other component outside and use it (FormRank in your case).